include/LightGBM/c_api.h

```diff
@@ -206,7 +209,9 @@ LIGHTGBM_C_EXPORT int LGBM_DatasetCreateFromCSC(const void* col_ptr,
                                                 int col_ptr_type,
                                                 const int32_t* indices,
                                                 const void* data,
+                                                const void* label,
                                                 int data_type,
+                                                int label_type,
```

Comment:
I remember we fix the label type to `double`. Maybe using the dynamic type is not needed.

Reply:
label type to double directly?

Reply:
The native floating point number of R only supports 64-bit, so I think keep `label_type` is necessary to support both R and python interfaces.